### PR TITLE
[gui] Support set_image with texture

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
This PR supported `set_image` with textures.

### CODE REVIEW

File: python/taichi/ui/staging_buffer.py [hunk 1]
```diff
@@ -122,7 +134,9 @@ def copy_image_u8_to_rgba8(src: ti.template(), dst: ti.template(),
 
 
 def to_rgba8(image):
-    if not hasattr(image, 'n') or image.m != 1:
+    is_texture = isinstance(image, Texture)
+
+    if not is_texture and (not hasattr(image, 'n') or image.m != 1):
         raise Exception(
             'the input image needs to be a Vector field (matrix with 1 column)'
```
Comment (ailzhang) [inline python/taichi/ui/staging_buffer.py:141]: update this error message as well? ;)

**Review by ailzhang — APPROVED** [review]
nit: can we also add a test without showing ?

**Comment by qiao-bo** [comment]
@PENGUINLIONG @ailzhang Just curious, is this `x.py` intended to be there?(top level on master) ;// and the reason is?

**Comment by PENGUINLIONG** [comment]
> @PENGUINLIONG @ailzhang Just curious, is this `x.py` intended to be there?(top level on master) ;// and the reason is?

@qiao-bo oof no，plz del it 🤡

**Pull request merged**